### PR TITLE
doc: Instruction for building p4-fusion locally.

### PR DESCRIPTION
**Pull request by ryanslade**
Instructions on how to build `p4-fusion` locally.

Closes https://github.com/sourcegraph/sourcegraph/issues/44762

## Test plan

Built locally

### CODE REVIEW

File: doc/dev/background-information/index.md [hunk 1]
```diff
@@ -38,6 +38,7 @@
 - [Developing an out-of-band migration](oobmigrations.md)
 - [Developing a background routine](backgroundroutine.md)
 - [Code host connections on local dev environment](code-host.md)
+- [Building p4-fusion](./build_p4_fusion.md)
```
Comment (mrnugget) [inline doc/dev/background-information/index.md:41]: This should also be linked on `doc/dev/index.md`:

https://github.com/sourcegraph/sourcegraph/blob/8869a0d16b9420519f75a9226885e799295c4a0b/doc/dev/index.md#L104

File: doc/dev/background-information/build_p4_fusion.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+In order to import Perforce depots into Sourcegraph we first convert them into git repositories. We use an open source tool called [p4-fusion](https://github.com/salesforce/p4-fusion):
+
+> A fast Perforce depot to Git repository converter using the Helix Core C/C++ API as an attempt to mitigate the performance bottlenecks in git-p4.py.
+
+[Building](https://github.com/salesforce/p4-fusion#build) p4-fusion can be a little tricky as it depends on some older libraries and also doesn't build on M1 Apple laptops. To get around this we use [nix](https://nixos.org).
+
+## How to build
+
+Below are the instructions for building p4-fusion locally, assuming you have the [Sourcegraph repo](https://github.com/sourcegraph/sourcegraph) checked out.
```
Comment (sashaostrikov) [inline doc/dev/background-information/build_p4_fusion.md:11]: To make it a bit less "slangy", wdyt?

```suggestion
Below are the instructions for building p4-fusion locally, assuming you have the [Sourcegraph repository](https://github.com/sourcegraph/sourcegraph) checked out.
```

**Review by sashaostrikov — APPROVED** [review]
LGTM!
Added 1 suggestion.

**Review by mrnugget — APPROVED** [review]
approve, but I think we should add that link

**Pull request merged**